src/App.tsx

```diff
@@ -1,24 +1,195 @@
+        setIsLoadingTodo(false);
+      });
+  };
+
+  const removeTodo = (id: number) => {
+    setIsLoadingTodo(true);
+    deleteTodo(id)
+      .then(() => {
+        const newTodosList = todos.filter(todo => todo.id !== id);
+
+        setTodos(newTodosList);
+        setIsLoadingTodo(false);
+      })
+      .catch(() => {
+        setErrorMessage('Unable to delete a todo');
+        setIsLoadingTodo(false);
+      });
+  };
+
+  const handleUpdateTodo = (todo: Todo) => {
+    updateTodo(todo.id, { completed: !todo.completed })
+      .then((updatedTodo: Todo) => {
+        setTodos(currentTodos => {
+          const todoIndex = todos
+            .findIndex(currTodo => currTodo.id === updatedTodo.id);
+
+          if (todoIndex > -1) {
+            return [
+              ...todos.slice(0, todoIndex),
+              updatedTodo,
+              ...todos.slice(todoIndex + 1),
+            ];
+          }
+
+          return currentTodos;
+        });
+      })
+      .catch(() => {
+        setErrorMessage('Unable to update a todo');
+      });
+  };
+
+  const filteredTodos: Todo[] = useMemo(() => {
+    return todos.filter((todo) => {
+      switch (currentFilter) {
+        case FilterType.Active:
+          return !todo.completed;
+        case FilterType.Completed:
+          return todo.completed;
+        default:
+          return true;
+      }
+    });
+  }, [todos, currentFilter]);
+
+  const completedTodos = todos.filter(todo => todo.completed);
+  const activeTodos = todos.filter(todo => !todo.completed);
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {activeTodos.length > 0
+            && (
+              <button type="button" className="todoapp__toggle-all active" />
+            )}
+
+          <TodoForm
+            setTodoTitle={setTodoTitle}
+            todoTitle={todoTitle}
+            onAdd={addNewTodo}
+          />
+        </header>
+
+        <section className="todoapp__main">
+          <TodoList
+            todos={filteredTodos}
+            onDelete={removeTodo}
+            onUpdateTodo={handleUpdateTodo}
+          />
+          {isLoadingTodo && <p className="loader" />}
+          {tempTodo && (
+            <TodoInfo
+              todo={tempTodo}
+              onDelete={removeTodo}
+              onUpdateTodo={handleUpdateTodo}
+            />
+          )}
+        </section>
+
+        {todos.length > 0
+          && (
+            <footer className="todoapp__footer">
```

Comment:
Think about how you can split all this logic. Maybe you can create separate components for separate logical parts?